.github/actions/create-pull-request/action.yml

```diff
@@ -0,0 +1,355 @@
+              core.setFailed(`Failed to get existing pull request reference: ${err}`);
+              core.error(err);
+              process.exit(1);
+            }
+            core.info("Existing pull request reference not found");
+          }
+
+          try {
+            core.info(`Creating new pull request reference:
+              owner: ${context.repo.owner}
+              repo:  ${context.repo.repo}
+              ref:   ${pullRequestFullRef}
+              sha:   ${context.sha}
+            `);
+
+            const { data: newRef } = await github.rest.git.createRef({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              ref: pullRequestFullRef,
+              sha: context.sha,
+            });
+
+            return newRef.object.sha;
+          } catch (err) {
+            core.setFailed(
+              `Failed to create/update pull request branch reference: ${err}`
+            );
+            core.error(err);
+          }
+
+    # Commit files using the GitHub API to ensure commits are signed
+    - name: 'Create Commits'
+      id: 'create-commits'
+      uses: 'actions/github-script@98814c53be79b1d30f795b907e553d8679345975' # ratchet:actions/github-script@v6
+      env:
+        HEAD_BRANCH: '${{ inputs.head_branch }}'
+        BASE_BRANCH: '${{ inputs.base_branch }}'
+        PR_TITLE: '${{ inputs.pull_request_title }}'
+        PR_BODY: '${{ inputs.pull_request_body }}'
+        CHANGED_PATHS: '${{ inputs.changed_paths }}'
+        DELETED_PATHS: '${{ inputs.deleted_paths }}'
+      with:
+        github-token: '${{ inputs.token }}'
+        retries: '${{ inputs.max_retries }}'
+        script: |-
+          try {
+            const fs = require("fs/promises");
+
+            const parentSHA = "${{ steps.base-branch-ref.outputs.result }}";
+            const pullRequestPartialRef = `heads/${process.env.HEAD_BRANCH}`;
+
+            // documented here: https://docs.github.com/en/rest/git/trees?apiVersion=2022-11-28#create-a-tree
+            const FILE_MODE = "100644";
+            const EXEC_MODE = "100755";
+
+            const prCommitTree = [];
+
+            const changedPaths = JSON.parse(process.env.CHANGED_PATHS);
+            const deletedPaths = JSON.parse(process.env.DELETED_PATHS);
+
+            // iterate the files loading their content into each object
+            await Promise.allSettled(
+              changedPaths.map(async (file) => {
+                const content = await fs.readFile(file, { encoding: "utf8" });
+                const isExec = !!((await fs.stat(file).mode) & fs.constants.S_IXUSR);
+                prCommitTree.push({
+                  path: file,
+                  mode: isExec ? EXEC_MODE : FILE_MODE,
+                  type: "blob",
+                  content: content,
+                });
+              })
+            );
+
+            // iterate the files loading their content into each object
+            await Promise.allSettled(
```

Comment:
I don't think `allSettled` will throw an error on rejection

Reply:
Hmm, i thought it would wait and throw error if any failed, read the docs wrong.

Updated.